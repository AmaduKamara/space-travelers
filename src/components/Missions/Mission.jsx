import React from 'react';
import PropTypes from 'prop-types';

const Mission = ({
  id, description, missionName, toggleMission, reserved,
}) => (
  <>
    <tbody className="odd:bg-white even:bg-slate-200 border-2 border-slate-100">
      <tr>
        <td className="px-6 py-4  whitespace-wrap flex align-top text-sm font-bold text-gray-900">{missionName}</td>
        <td className="text-sm border-2 border-slate-100 text-gray-900 tracking-tight font-medium px-6 py-4 ">
          {description}
        </td>
        <td className="text-sm border-2 border-slate-100 text-gray-900 font-light px-6 py-4 whitespace-nowrap">
          {reserved ? <span className="text-xs px-2 py-0.5 font-bold bg-blue-500 text-white rounded">Active Member</span> : <span className="text-xs px-2 py-0.5 font-bold bg-gray-600 text-white rounded">NOT A MEMBER</span>}
        </td>
        <td className="text-sm text-gray-900 font-light px-6 py-4 whitespace-nowrap">
          {reserved ? <button type="button" id={id} onClick={toggleMission} className="bg-transparent hover:border-red-900 text-red-400 font-bold py-2 px-4 border border-red-400 rounded">Leave Mission</button> : <button type="button" id={id} onClick={toggleMission} className="bg-transparent hover:border-black font-bold py-2 px-4 border border-gray-400 text-black rounded">Join Mission</button>}

        </td>
      </tr>
    </tbody>
  </>
);

export default Mission;

Mission.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  missionName: PropTypes.string.isRequired,
  toggleMission: PropTypes.func.isRequired,
  reserved: PropTypes.bool,

};

Mission.defaultProps = {
  reserved: false,
};
