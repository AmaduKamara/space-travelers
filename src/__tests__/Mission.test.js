import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import Mission from '../components/Missions/Mission';

const renderer = require('react-test-renderer');

it('renders the button toggle true state correctly', () => {
  const toggleMission = () => true;
  const { getByTestId } = render(<Mission
    toggleMission={toggleMission}
    id="2"
    description="Render Description"
    missionName="Render Missiom"
    reserved
  />);
  expect(getByTestId('mission-btn')).toHaveTextContent('Leave Mission');
});

it('renders the button toggle false state correctly', () => {
  const toggleMission = () => false;
  const { getByTestId } = render(<Mission
    toggleMission={toggleMission}
    id="2"
    description="Render Description"
    missionName="Render Missiom"
    reserved={false}
  />);
  expect(getByTestId('mission-btn')).toHaveTextContent('Join Mission');
});

it('matches snapshot of Misson component', () => {
  const toggleMission = () => true;
  const id = '1';
  const description = 'test';
  const missionName = 'test mission';
  const reserved = true;
  const tree = renderer.create(<Mission
    toggleMission={toggleMission}
    id={id}
    description={description}
    missionName={missionName}
    reserved={reserved}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
