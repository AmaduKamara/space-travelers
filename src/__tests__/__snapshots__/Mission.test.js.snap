// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of Misson component 1`] = `
<tbody
  className="odd:bg-white even:bg-slate-200 border-2 border-slate-100"
>
  <tr>
    <td
      className="px-6 py-4  whitespace-wrap flex align-top text-sm font-bold text-gray-900"
    >
      test mission
    </td>
    <td
      className="text-sm border-2 border-slate-100 text-gray-900 tracking-tight font-medium px-6 py-4 "
    >
      test
    </td>
    <td
      className="text-sm border-2 border-slate-100 text-gray-900 font-light px-6 py-4 whitespace-nowrap"
    >
      <span
        className="text-xs px-2 py-0.5 font-bold bg-sky-500 text-white rounded"
      >
        Active Member
      </span>
    </td>
    <td
      className="text-sm text-gray-900 font-light px-6 py-4 whitespace-nowrap"
      data-testid="mission-btn"
    >
      <button
        className="bg-transparent hover:border-red-900 text-red-400 font-bold py-2 px-4 border border-red-400 rounded"
        id="1"
        onClick={[Function]}
        type="button"
      >
        Leave Mission
      </button>
    </td>
  </tr>
</tbody>
`;
